spec/lib/fat_zebra/authorise_spec.rb

```diff
@@ -0,0 +1,103 @@
+
+describe FatZebra::Authorise do
+
+  describe '.token', :vcr do
+    subject(:token) { FatZebra::Authorise.token(valid_3ds_token_payload) }
+
+    let(:valid_3ds_token_payload) {{
+      reference_id: 'xxxx-xxxx-xxxx',
+      confirm_url: 'SOME URL',
+      order_number: 'xxxx-xxxx-xxxx',
+      currency_code: '036',
+      amount: 100
+    }}
+
+    it 'returns JWT with given params' do
+      is_expected.to be_accepted
+      expect(token.jwt).to be_truthy
+      expect(token.reference_id).to be_truthy
+    end
+
+    it 'returns JWT with some params' do
+      valid_3ds_token_payload.delete(:reference_id)
+      is_expected.to be_accepted
+      expect(token.jwt).to be_truthy
+      expect(token.reference_id).to be_truthy
+    end
+
+    it 'returns JWT with no params' do
+      valid_3ds_token_payload.clear
+      is_expected.to be_accepted
+      expect(token.jwt).to be_truthy
+      expect(token.reference_id).to be_truthy
+    end
+  end
+
+  describe '.authorise', :vcr do
+    subject(:authorise) { FatZebra::Authorise.authorise(valid_3ds_authorise_payload) }
+
+    context 'with invalid input' do
+      before do
+        valid_3ds_authorise_payload[:card_expiry] = DateTime.now.prev_month.strftime('%m/%Y')
+      end
+
+      it do
+        is_expected.not_to be_accepted
+        expect(authorise.errors).to match(/Error Validating Credit Card Expiration/)
+      end
+    end
+
+    context 'with frictionless response' do
+      before do
+        valid_3ds_authorise_payload[:card_number] = '4000000000001000'
+      end
+
+      it do
+        is_expected.to be_accepted
+        expect(authorise.keys).to include('Enrolled', 'ACSUrl', 'CardBin', 'TransactionId')
+        expect(authorise.errors).not_to be_truthy
+      end
+
+      it do
+        expect(authorise.Enrolled).to eq('Y')
+        expect(authorise.ACSUrl).not_to be_truthy
+        expect(authorise.CardBin).to be_truthy
+        expect(authorise.ErrorNo).to eq('0')
+        expect(authorise.ErrorDesc).not_to be_truthy
+        expect(authorise.CardBrand).to be_truthy
+        expect(authorise.TransactionId).to be_truthy
+      end
+    end
+
+    context 'with challenge response' do
+      before do
+        valid_3ds_authorise_payload[:card_number] = '4111111111111111'
+      end
+
+      it do
+        is_expected.to be_accepted
+        expect(authorise.keys).to include('Enrolled', 'ACSUrl', 'CardBin', 'TransactionId')
+        expect(authorise.errors).not_to be_truthy
+      end
+      
+      it do
+        expect(authorise.Enrolled).to eq('Y')
+        expect(authorise.ACSUrl).to be_truthy
+        expect(authorise.CardBin).to be_truthy
+        expect(authorise.ErrorNo).to eq('0')
+        expect(authorise.ErrorDesc).not_to be_truthy
+        expect(authorise.CardBrand).to be_truthy
+        expect(authorise.TransactionId).to be_truthy
+      end
+    end
+    
+    context 'validations' do
+      before do
+        valid_3ds_authorise_payload.clear
```

Comment:
Minor suggestion, could remove this block and use an overriding `let` instead:
```ruby
let(:valid_3ds_token_payload) { {} }
```

Reply:
:+1